File: splearn/src/test/java/com/splearn/application/required/MemberRepositoryTest.java
package com.splearn.application.required;

import com.splearn.domain.Member;
import com.splearn.domain.MemberFixture;
import jakarta.persistence.EntityManager;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import static com.splearn.domain.MemberFixture.createMemberRegisterRequest;
import static com.splearn.domain.MemberFixture.createPasswordEncoder;
import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.*;

@DataJpaTest
class MemberRepositoryTest {
    @Autowired
    MemberRepository memberRepository;

    @Autowired
    EntityManager entityManager;

    @Test
    void createMember(){
        var member = Member.register(createMemberRegisterRequest(), createPasswordEncoder());

        memberRepository.save(member);

        entityManager.flush();
    }


}

File: splearn/src/test/kotlin/cham/splearn/application/required/MemberRepositoryTest.kt
package cham.splearn.application.required

import cham.splearn.domain.Member
import cham.splearn.domain.createPasswordEncoder
import cham.splearn.domain.createMemberRegisterRequest
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldNotBe
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.extensions.spring.SpringExtension
import jakarta.persistence.EntityManager
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.dao.DataIntegrityViolationException

@DataJpaTest
class MemberRepositoryTest : FunSpec() {
    
    override fun extensions() = listOf(SpringExtension)
    
    @Autowired
    lateinit var memberRepository: MemberRepository

    @Autowired
    lateinit var entityManager: EntityManager

    init {
        test("createMember") {
            val member = memberRepository.save(Member.register(createMemberRegisterRequest(), createPasswordEncoder()))

            member.id shouldNotBe null

            entityManager.flush()
        }

        test("duplicateEmailFail") {
            val member1 = memberRepository.save(Member.register(createMemberRegisterRequest(), createPasswordEncoder()))

            val member2 = Member.register(createMemberRegisterRequest(), createPasswordEncoder())

            shouldThrow<DataIntegrityViolationException> {
                memberRepository.save(member2)
            }
        }
    }
}